packages/foreman/rubygem-rest-client/rubygem-rest-client.spec

```diff
@@ -1,83 +1,112 @@
 Requires: %{?scl_prefix}rubygem(netrc) >= 0.8
-Requires: %{?scl_prefix}rubygem(netrc) < 1.0
+Requires: %{?scl_prefix}rubygem(netrc) < 1
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby >= 2.0.0
 BuildRequires: %{?scl_prefix_ruby}rubygems-devel
 BuildArch: noarch
 Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
-%{?scl:Obsoletes: ruby193-rubygem-%{gem_name}}
+# end specfile generated dependencies
 
 %description
-A simple Simple HTTP and REST client for Ruby, inspired by the Sinatra
-microframework style of specifying actions: get, put, post, delete.
+A simple HTTP and REST client for Ruby, inspired by the Sinatra microframework
+style of specifying actions: get, put, post, delete.
+
 
 %package doc
-BuildArch:  noarch
-Requires:   %{?scl_prefix}%{pkg_name} = %{version}-%{release}
-Summary:    Documentation for rubygem-%{gem_name}
+Summary: Documentation for %{pkg_name}
+Group: Documentation
+Requires: %{?scl_prefix}%{pkg_name} = %{version}-%{release}
+BuildArch: noarch
 
 %description doc
-This package contains documentation for rubygem-%{gem_name}.
+Documentation for %{pkg_name}.
 
 %prep
-%setup -n %{pkg_name}-%{version} -T -c
-%{?scl:scl enable %{scl} - <<EOF}
-%gem_install -n %{SOURCE0}
+%{?scl:scl enable %{scl} - << \EOF}
+gem unpack %{SOURCE0}
+%{?scl:EOF}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+%{?scl:scl enable %{scl} - << \EOF}
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
 %{?scl:EOF}
 
 %build
+# Create the gem as gem install only works on a gem file
+%{?scl:scl enable %{scl} - << \EOF}
+gem build %{gem_name}.gemspec
+%{?scl:EOF}
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%{?scl:scl enable %{scl} - << \EOF}
+%gem_install
+%{?scl:EOF}
 
 %install
 mkdir -p %{buildroot}%{gem_dir}
-cp -a .%{gem_dir}/* \
+cp -pa .%{gem_dir}/* \
         %{buildroot}%{gem_dir}/
 
 mkdir -p %{buildroot}%{_bindir}
-cp -a .%{_bindir}/* \
+cp -pa .%{_bindir}/* \
         %{buildroot}%{_bindir}/
-
 find %{buildroot}%{gem_instdir}/bin -type f | xargs chmod a+x
 
 %files
-%defattr(-, root, root, -)
-%{_bindir}/restclient
 %dir %{gem_instdir}
+%{_bindir}/restclient
+%exclude %{gem_instdir}/.gitignore
+%exclude %{gem_instdir}/.rubocop-disables.yml
+%exclude %{gem_instdir}/.rubocop.yml
+%exclude %{gem_instdir}/.travis.yml
+%license %{gem_instdir}/LICENSE
 %{gem_instdir}/bin
 %{gem_libdir}
-%doc %{gem_instdir}/LICENSE
 %exclude %{gem_cache}
 %{gem_spec}
 
 %files doc
 %doc %{gem_docdir}
+%exclude %{gem_instdir}/.rspec
 %doc %{gem_instdir}/AUTHORS
+%{gem_instdir}/Gemfile
 %doc %{gem_instdir}/README.md
+%{gem_instdir}/Rakefile
 %doc %{gem_instdir}/history.md
-
-%exclude %{gem_instdir}/.*
-%exclude %{gem_instdir}/*.gemspec
-%exclude %{gem_instdir}/Gemfile
-%exclude %{gem_instdir}/spec
-%exclude %{gem_instdir}/Rakefile
+%{gem_instdir}/rest-client.gemspec
+%{gem_instdir}/rest-client.windows.gemspec
```

Comment:
Should we exclude this?

Reply:
The windows gemspec you mean? could, yes.